packages/builder/src/components/automation/SetupPanel/FieldSelector.svelte

```diff
@@ -38,77 +44,58 @@
     NUMBER: null,
     DATETIME: null,
     STRING: "",
-    OPTIONS: [],
-    ARRAY: [],
+    ARRAY: "",
   }
 
-  const coerce = (value, type) => {
-    const re = new RegExp(/{{([^{].*?)}}/g)
-    if (re.test(value)) {
-      return value
-    }
-
-    if (type === "boolean") {
-      if (typeof value === "boolean") {
-        return value
-      }
-      return value === "true"
-    }
-    if (type === "number") {
-      if (typeof value === "number") {
-        return value
-      }
-      return Number(value)
-    }
-    if (type === "options") {
-      return [value]
-    }
-    if (type === "array") {
-      if (Array.isArray(value)) {
-        return value
-      }
-      return value.split(",").map(x => x.trim())
-    }
-
-    if (type === "link") {
-      if (Array.isArray(value)) {
-        return value
-      }
-
-      return [value]
+  const onChange = (e, field) => {
+    if (e.detail !== editableValue[field]) {
+      editableValue[field] = e.detail
+      dispatch("change", editableValue)
     }
-
-    return value
-  }
-
-  const onChange = (e, field, type) => {
-    value[field] = coerce(e.detail, type)
-    dispatch("change", value)
   }
 </script>
 
 {#if schemaFields.length && isTestModal}
-  <div class="schema-fields">
+  <div class="fields">
     {#each schemaFields as [field, schema]}
-      <RowSelectorTypes
-        {isTestModal}
-        {field}
-        {schema}
-        {bindings}
-        {value}
-        {onChange}
-      />
+      <PropField label={field}>
+        {#if ["string", "number", "array"].includes(schema.type)}
```

Comment:
NAB but might be nice to use the `FieldType` enums here rather than magic strings for the data types.